lib/src/genetic_algorithm.dart

```diff
@@ -50,39 +51,40 @@ class GeneticAlgorithm<P extends Phenotype<G, R>, G, R extends FitnessResult> {
   /// Function used for printing info about the progress of the genetic
   /// algorithm. This is the standard console [print] by default.
   final PrintFunction printf;
+
   /// Function used for showing status of the genetic algorithm, with the
   /// assumption that previous status text is rewritten by new status text.
   /// This is also initialized with [print] by default, but that's not
   /// the ideal implementation.
   final PrintFunction statusf;
 
   void _evaluateNextGeneration() {
-    evaluateLastGeneration().then((Object _) {
-      printf("Applying niching to results.");
-      breeder.applyFitnessSharingToResults(generations.last);
-      printf("Generation #$currentGeneration evaluation done. Results:");
+    evaluateLastGeneration().then<void>((dynamic _) {
```

Comment:
Maybe just `_` since it's `void` anyway?

```suggestion
    evaluateLastGeneration().then<void>((_) {
```

Reply:
Turns out, not so much.

<img width="566" alt="Screen Shot 2021-04-13 at 12 39 40 pm" src="https://user-images.githubusercontent.com/30503/114488854-71c7ba00-9c55-11eb-945d-e45054e11e4a.png">

The required type of the function is `void Function(dynamic)`. Oh wells. =)